library/src/main/java/com/google/maps/android/data/kml/KmlLayer.java

```diff
@@ -88,11 +100,13 @@ public KmlLayer(GoogleMap map, int resourceId, FragmentActivity activity, Marker
     /**
      * Creates a new KmlLayer object - addLayerToMap() must be called to trigger rendering onto a map.
      *
+     * Constructor may be called on a background thread, as I/O and parsing may be long-running.
+     *
      * Use this constructor with shared object managers in order to handle multiple layers with
      * their own event handlers on the map.
      *
      * @param map    GoogleMap object
-     * @param stream InputStream containing KML file
+     * @param stream InputStream containing KML or KMZ file
```

Comment:
I think it would be simpler if another constructor was created, with the following signature, to handle the KMZ files explicitly:

```java
public KmlLayer(GoogleMap map, ZipInputStream stream, FragmentActivity activity, MarkerManager markerManager, PolygonManager polygonManager, PolylineManager polylineManager, GroundOverlayManager groundOverlayManager)
```

This way, it's left to the caller to use the correct constructor for their use-case.

Reply:
I thought about that. Since we can differentiate between the two types with the logic I implemented, I thought a single universal constructor was simpler for the caller to use. Also, since there are two constructors currently, the other supports opening a KML by resource ID, how could we implement opening a KMZ by resource ID? We'd have to imply the resource ID is either a KML or KMZ or we'd need the same logic to differentiate between them and call the appropriate stream constructor anyway.

Reply:
Good point about invoking by resource ID. Let's stick to this implementation.